qiskit/circuit/library/n_local/n_local.py

```diff
@@ -1011,7 +1011,12 @@ def get_entangler_map(
 
     if entanglement == "full":
         return list(combinations(list(range(n)), m))
-    if entanglement in ["linear", "circular", "sca", "pairwise"]:
+    elif entanglement == "reverse_linear":
+        # reverse linear connectivity. In the case of m=2 and the entanglement_block='cx'
+        # then it's equivalent to 'full' entanglement
+        reverse = [tuple(range(n - i - m, n - i - m + 2)) for i in range(n - m + 1)]
```

Comment:
It looks like this only works for blocksizes of `m=2`. Should this be 
```suggestion
        reverse = [tuple(range(n - i - m, n - i)) for i in range(n - m + 1)]
```
to allow arbitrary block sizes? Could you add a test for different block sizes than 2 (e.g. 1 and 3) to make sure that works?